src/main/java/edu/kis/powp/jobs2d/drivers/DriversComposite.java

```diff
@@ -1,58 +1,90 @@
 package edu.kis.powp.jobs2d.drivers;
 
+import edu.kis.powp.jobs2d.Job2dDriver;
 import edu.kis.powp.jobs2d.drivers.visitor.IDriverVisitor;
 import edu.kis.powp.jobs2d.drivers.visitor.IVisitableDriver;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.*;
 import java.util.stream.Collectors;
 
 public class DriversComposite implements IVisitableDriver {
+    private final Map<Integer, IVisitableDriver> map;
+    private DriverManager driverManager;
 
-    private List<IVisitableDriver> list;
+    public DriversComposite(DriverManager driverManager) {
+        this.map = new TreeMap<>();
+        this.driverManager = driverManager;
+    }
 
-    public DriversComposite() {
-        this.list = new ArrayList<>();
+    public DriversComposite(Map<Integer, IVisitableDriver> list, DriverManager driverManager) {
+        this.map = list;
+        this.driverManager = driverManager;
     }
 
-    public DriversComposite(List<IVisitableDriver> list) {
-        this.list = list;
+    public void addDriver(int position, IVisitableDriver driver) {
+        this.map.put(position, driver);
     }
 
-    public void addDriver(IVisitableDriver driver) {
-        this.list.add(driver);
+    public boolean removeDriver(int position) {
+        return map.remove(position) != null;
     }
 
-    public boolean removeDriver(IVisitableDriver driver) {
-        return list.remove(driver);
+    public Map<Integer, IVisitableDriver> getMap() {
+        return map;
     }
 
     @Override
     public void setPosition(int x, int y) {
-        for (IVisitableDriver driver : list) {
-            driver.setPosition(x, y);
+        int startXAtStart = driverManager.getCurrentDriver().getStartX();
+        int startYAtStart = driverManager.getCurrentDriver().getStartY();
+        int counter = 0;
+        for (Job2dDriver driver : map.values()) {
+            if (counter == 0) {
+                driver.setPosition(x, y);
+            }
+            else {
+                int startX = driverManager.getCurrentDriver().getStartX();
+                int startY = driverManager.getCurrentDriver().getStartY();
+                driverManager.getCurrentDriver().setStartX(startXAtStart);
+                driverManager.getCurrentDriver().setStartY(startYAtStart);
+                driver.setPosition(startX, startY);
+            }
```

Comment:
This part is not intuitive. It is not clear why: 
driver.setPosition(x, y);
could not be applied to all children of the composite.